pinot-controller/src/main/java/org/apache/pinot/controller/util/TableMetadataReader.java

```diff
@@ -55,6 +61,27 @@ public TableMetadataReader(Executor executor, HttpClientConnectionManager connec
     _pinotHelixResourceManager = helixResourceManager;
   }
 
+  public Map<String, JsonNode> getServerCheckSegmentsReloadMetadata(String tableName, TableType tableType,
+      int timeoutMs)
+      throws InvalidConfigException, IOException {
+    String tableNameWithType =
```

Comment:
This check should be performed in the `RestletResource` class before calling this method